Deno.Command#spawn stdout/err pipes should be `inherit` by default?

This was surprising to me:

```ts
const command = new Deno.Command(Deno.execPath(), {
  args: [
    "eval",
    "--unstable",
    "--no-check",
    code,
  ],
  stdout: "piped",
});

const child = command.spawn();
// etc...
```

Then in the test output:

```ts
error: AssertionError: Test case is leaking 1 resource:

 - A child process stderr (rid 90) was opened during the test, but not closed during the test. Close the child process stderr by calling `proc.stderr.close()` or `await child.stderr.cancel()`.

    at assert (deno:ext/web/00_infra.js:295:13)
    at resourceSanitizer (deno:cli/js/40_testing.js:413:7)
    at async exitSanitizer (deno:cli/js/40_testing.js:431:9)
    at async Object.applyPermissions [as fn] (deno:cli/js/40_testing.js:533:9)
    at async runTest (deno:cli/js/40_testing.js:836:7)
    at async runTests (deno:cli/js/40_testing.js:1091:22)
```

I would have expected that if I don't set `stderr` to `"piped"` explicitly, then it would be "inherit". I don't think "piped" should be the default in this case because I believe it could lead to a deadlock if `stderr`'s buffer fills up since nothing is reading from it. Plus, in this case it means I have to explicitly handle it.

cc @crowlKats, what is the correct behaviour here?

--- a/cli/tests/unit/command_test.ts
+++ b/cli/tests/unit/command_test.ts
@@ -91,6 +91,7 @@ Deno.test(
         "await Deno.stdout.write(new TextEncoder().encode('hello'))",
       ],
       stderr: "null",
+      stdout: "piped",
     });
     const child = command.spawn();
 
@@ -124,6 +125,7 @@ Deno.test(
         "await Deno.stderr.write(new TextEncoder().encode('hello'))",
       ],
       stdout: "null",
+      stderr: "piped",
     });
     const child = command.spawn();
 
@@ -163,6 +165,8 @@ Deno.test(
         "eval",
         "Deno.stderr.write(new TextEncoder().encode('error\\n')); Deno.stdout.write(new TextEncoder().encode('output\\n'));",
       ],
+      stdout: "piped",
+      stderr: "piped",
     });
     const child = command.spawn();
     await child.stdout.pipeTo(file.writable, {

--- a/cli/tests/unit/flock_test.ts
+++ b/cli/tests/unit/flock_test.ts
@@ -151,6 +151,8 @@ function runFlockTestProcess(opts: { exclusive: boolean; sync: boolean }) {
   const process = new Deno.Command(Deno.execPath(), {
     args: ["eval", "--unstable", scriptText],
     stdin: "piped",
+    stdout: "piped",
+    stderr: "null",
   }).spawn();
 
   const waitSignal = async () => {

--- a/cli/tests/unit/http_test.ts
+++ b/cli/tests/unit/http_test.ts
@@ -1383,6 +1383,7 @@ Deno.test({
       const { success, stdout } = await new Deno.Command("curl", {
         args,
         stderr: "null",
+        stdout: "piped",
       }).output();
       assert(success);
       const output = decoder.decode(stdout);
@@ -1430,7 +1431,11 @@ Deno.test({
         "--header",
         "Accept-Encoding: gzip, deflate, br",
       ];
-      const proc = new Deno.Command("curl", { args, stderr: "null" }).spawn();
+      const proc = new Deno.Command("curl", {
+        args,
+        stderr: "null",
+        stdout: "piped",
+      }).spawn();
       const status = await proc.status;
       assert(status.success);
       const stdout = proc.stdout
@@ -1488,6 +1493,7 @@ Deno.test({
       const { success, stdout } = await new Deno.Command("curl", {
         args,
         stderr: "null",
+        stdout: "piped",
       }).output();
       assert(success);
       const output = decoder.decode(stdout).toLocaleLowerCase();
@@ -1543,6 +1549,7 @@ Deno.test({
       const { success, stdout } = await new Deno.Command("curl", {
         args,
         stderr: "null",
+        stdout: "piped",
       }).output();
       assert(success);
       const output = decoder.decode(stdout);
@@ -1595,6 +1602,7 @@ Deno.test({
       const { success, stdout } = await new Deno.Command("curl", {
         args,
         stderr: "null",
+        stdout: "piped",
       }).output();
       assert(success);
       const output = decoder.decode(stdout);
@@ -1651,6 +1659,7 @@ Deno.test({
       const { success, stdout } = await new Deno.Command("curl", {
         args,
         stderr: "null",
+        stdout: "piped",
       }).output();
       assert(success);
       const output = decoder.decode(stdout);
@@ -1709,6 +1718,7 @@ Deno.test({
       const { success, stdout } = await new Deno.Command("curl", {
         args,
         stderr: "null",
+        stdout: "piped",
       }).output();
       assert(success);
       const output = decoder.decode(stdout);
@@ -1767,6 +1777,7 @@ Deno.test({
       const { success, stdout } = await new Deno.Command("curl", {
         args,
         stderr: "null",
+        stdout: "piped",
       }).output();
       assert(success);
       const output = decoder.decode(stdout);
@@ -1822,6 +1833,7 @@ Deno.test({
       const { success, stdout } = await new Deno.Command("curl", {
         args,
         stderr: "null",
+        stdout: "piped",
       }).output();
       assert(success);
       const output = decoder.decode(stdout);
@@ -1883,6 +1895,7 @@ Deno.test({
       const { success, stdout } = await new Deno.Command("curl", {
         args,
         stderr: "null",
+        stdout: "piped",
       }).output();
       assert(success);
       const output = decoder.decode(stdout);
@@ -1939,7 +1952,11 @@ Deno.test({
         "--header",
         "Accept-Encoding: gzip, deflate, br",
       ];
-      const proc = new Deno.Command("curl", { args, stderr: "null" }).spawn();
+      const proc = new Deno.Command("curl", {
+        args,
+        stderr: "null",
+        stdout: "piped",
+      }).spawn();
       const status = await proc.status;
       assert(status.success);
       const stdout = proc.stdout
@@ -2007,6 +2024,7 @@ Deno.test({
       const { success, stdout } = await new Deno.Command("curl", {
         args,
         stderr: "null",
+        stdout: "piped",
       }).output();
       assert(success);
       const output = decoder.decode(stdout);
@@ -2569,7 +2587,11 @@ Deno.test({
         "Accept-Encoding: gzip, deflate, br",
         "--no-buffer",
       ];
-      const proc = new Deno.Command("curl", { args, stderr: "null" }).spawn();
+      const proc = new Deno.Command("curl", {
+        args,
+        stderr: "null",
+        stdout: "piped",
+      }).spawn();
       const stdout = proc.stdout
         .pipeThrough(new DecompressionStream("gzip"))
         .pipeThrough(new TextDecoderStream());

--- a/cli/tsc/dts/lib.deno.unstable.d.ts
+++ b/cli/tsc/dts/lib.deno.unstable.d.ts
@@ -1456,11 +1456,13 @@ declare namespace Deno {
     stdin?: "piped" | "inherit" | "null";
     /**  How `stdout` of the spawned process should be handled.
      *
-     * Defaults to `"piped"`. */
+     * Defaults to `"piped"` for `output` & `outputSync`,
+     * and `"inherit"` for `spawn`. */
     stdout?: "piped" | "inherit" | "null";
     /**  How `stderr` of the spawned process should be handled.
      *
-     * Defaults to "piped". */
+     * Defaults to "piped" for `output` & `outputSync`,
+     * and `"inherit"` for `spawn`. */
     stderr?: "piped" | "inherit" | "null";
 
     /** Skips quoting and escaping of the arguments on windows. This option

--- a/runtime/js/40_spawn.js
+++ b/runtime/js/40_spawn.js
@@ -307,7 +307,12 @@
       }
 
       spawn() {
-        return spawnChild(this.#command, this.#options);
+        const options = {
+          ...(this.#options ?? {}),
+          stdout: this.#options?.stdout ?? "inherit",
+          stderr: this.#options?.stderr ?? "inherit",
+        };
+        return spawnChild(this.#command, options);
       }
     };
   }

